### PR TITLE
Add tool definitions for datamash.

### PR DESCRIPTION
**Pull request by davebx**

### CODE REVIEW

File: tools/datamash/datamash-ops.xml [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+                #end if
+                #for $oper in $operations
+                    ${oper.op_name}
+                    ${oper.op_column}
+                #end for
+                < $in_file > $out_file
+          ]]>
+    </command>
+    <expand macro="inputs_outputs">
+        <param help="Example: 1,4 - To group by the first and fourth fields. Leave empty to perform operation on entire file as one group." label="Group by fields" name="grouping" type="text">
```
Comment (nsoranzo) [inline tools/datamash/datamash-ops.xml]: `Example: 1,4 - To group by the first and fourth fields.`

I'd write this as:

`Example: to group by the first and fourth fields, use 1,4 .`

File: tools/datamash/datamash-ops.xml [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+            <validator message="invalid fields value" type="regex">^[0-9, ]*$</validator>
```
Comment (nsoranzo) [inline tools/datamash/datamash-ops.xml]: I'd specify the list of allowed values in the `message`.

File: tools/datamash/datamash-ops.xml [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+<![CDATA[
+.. class:: infomark
+
+**TIP:** Input data must be TAB delimited. If the desired dataset does not appear in the input list, use *Text Manipulation->Convert* to convert it to **Tabular** type.
+
+-----
+
+**Syntax**
+
+This tools performs common operations (such as summing, counting, mean, standard-deviation) on input file, based on tabular data. The tool can also optionaly group the input based on a given field.
```
Comment (nsoranzo) [inline tools/datamash/datamash-ops.xml]: s/optionaly/optionally/

File: tools/datamash/macros.xml [hunk 4]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0"?>
+<macros>
+    <xml name="inputs_outputs">
+        <inputs>
+            <param format="tabular" help="" label="Select Input Data" name="in_file" type="data" />
```
Comment (nsoranzo) [inline tools/datamash/macros.xml]: s/Select Input Data/Input tabular dataset/ ?

**Comment by jmchilton** [comment]
Why here and not IUC? I thought we were putting new tools in the IUC repository.

**Comment by davebx** [comment]
@jmchilton I'm not deeply invested in one over the other, if consensus says iuc, iuc it is.

**Comment by bgruening** [comment]
@davebx have you looked at https://toolshed.g2.bx.psu.edu/view/agordon/datamash_wrapper/

This one is working fine for us since a year or so. Might be worth to have a look.

**Comment by nsoranzo** [comment]
Closing in favour of https://github.com/galaxyproject/tools-iuc/issues/238 .

**Pull request closed without merging**